src/Compilers/Core/Desktop/CommandLine/CommonCommandLineParser.cs

```diff
@@ -612,9 +654,9 @@ private static IEnumerable<string> Split(string str, Func<char, bool> splitHere)
         }
 
         /// <summary>
-        /// Condense double backslashes that precede a quotation mark to single backslashes.
+        /// Make all backslashes escaped regardless whether they preceed a quote or not.
```

Comment:
That is correct. In the input, the backslashes not preceding quotes are not escaped (possibly to not force the user to type so many slashes). That makes it very hard to either unescape slashes after quotes are removed or remove quotes after slashes are unescaped.
What is happening here is that I make the slashes that do not precede quotes to be "escaped" the same way as the slashes that do. This way I can later unescape slashes and not rely on presence of quotes.